packages/kbn-openapi-generator/src/parser/lib/get_imports_map.ts

```diff
@@ -10,30 +10,67 @@ import { uniq } from 'lodash';
 import type { OpenApiDocument } from '../openapi_types';
 import { findRefs } from './find_refs';
 
+const SCHEMA_COMPONENT_PATH = '#/components/schemas/';
+
 export interface ImportsMap {
   [importPath: string]: string[];
 }
 
 /**
- * Traverse the OpenAPI document, find all external references, and return a map
- * of import paths and imported symbols
+ * Find all external schema references in the document,
+ * return a map of import paths and imported symbols
  *
  * @param parsedSchema Parsed OpenAPI document
  * @returns A map of import paths to symbols to import
  */
 export const getImportsMap = (parsedSchema: OpenApiDocument): ImportsMap => {
```

Comment:
I have refactored this function to use `reduce`, and split out the `parseRef` function, let me know what you think, I know some people prefer to avoid reduce.